### DISCUSSION TITLE
Unable to perform downgrade after updating to Wix4

### DISCUSSION THREAD

**Discussion by appu-george** [wixtoolset/issues#7756] [Questions] [answered]
I have a Wix installer project which was initially authored using Wix3 toolset and recently converted it to Wix4 using the conversion tool (heatwave extension in Visual Studio). I am using a managed bootstrapper application and some changes were required in the BA as part of the Wix4 upgrade. One thing that is not working is downgrade. Even though I have set `AllowDowngrades="yes"` in my package, the downgrade operation (eg: from version 1.0.0.10 -> 1.0.0.9) fails with error code 0x80070666 in the `Apply` phase. Is there anything specific that need to be done for downgrade to work in Wix4? I also see in the logs (attached herewith) that planning is skipped once it detected a downgrade.

[Installation_downgrade.txt](https://github.com/wixtoolset/issues/files/12728383/Installation_downgrade.txt)

**Comment by appu-george** [comment]
In an attempt to isolate the issue, I have tried replacing the custom bootstrapper with the standard bootstrapper application and I still get the same issue. The log is attached here.

[Installation_downgrade_stdba.txt](https://github.com/wixtoolset/issues/files/12738811/Installation_downgrade_stdba.txt)

**Comment by barnson** [comment, marked as answer]
That was an intentional design change in WiX v4. See [the PR and linked issues](https://github.com/wixtoolset/wix/pull/140) for details. Generally, MSI packages don't downgrade well (MSI likes to keep higher-versioned files) so the best approach is generally to have the BA uninstall the higher-versioned bundle.

**Reply by stvoorhees** [reply]
Is there any reference for how the uninstallation should be handled? I am writing my own uninstallation logic (against the higher-versioned bundle) and am encountering the same issue.
